models/models.go

```diff
@@ -114,6 +114,69 @@ func (filter *Filter) ValidateFilterBlueprint() error {
 	return nil
 }
 
+// DatasetDimensionResults represents a structure for a list of dimensions
+type DatasetDimensionResults struct {
+	Items []DatasetDimension `json:"items"`
+}
+
+// DatasetDimension represents an overview for a single dimension. This includes a link to the code list API
+// which provides metadata about the dimension and all possible values.
+type DatasetDimension struct {
+	Name string `bson:"name,omitempty"          json:"dimension,omitempty"`
+}
+
+// ValidateFilterDimensions checks the selected filter dimension
+// are valid for a version of a dataset
+func ValidateFilterDimensions(filterDimensions []Dimension, datasetDimensions *DatasetDimensionResults) error {
+	dimensionNames := make(map[string]string)
+	for _, datasetDimension := range datasetDimensions.Items {
+		dimensionNames[datasetDimension.Name] = datasetDimension.Name
```

Comment:
this map (and same in the following func), seems odd - wouldn't `dimensionNames[datasetDimension.Name] = 1` here...

Reply:
Agreed, rushing to get a quick solution out there :/